### ISSUE TITLE
Bun crashes when migrating from package-lock with a git dependency

### ISSUE THREAD

**Issue by wundersolutions-juanjo** [oven-sh/bun#12205]
### How can we reproduce the crash?

Tested with the following package.json, no other files on folder, running on a Powershell windows terminal
```json
{
  "dependencies": {
    "sass": "^1.35.1",
    "@hasanm95/cogo-toast": "https://github.com/hasanm95/cogo-toast"
  }
}
```

- `npm i`
- `Remove-Item -r .\node_modules\`
- `bun i`

Using NPM 10.2.4

### Relevant log output

Bun v1.1.13 (bd6a6051) Windows x64
Args: "C:\Users\Juanjo\.bun\bin\bun.exe" "i"
Elapsed: 2033ms | User: 15ms | Sys: 93ms
RSS: 97.37MB | Peak: 97.64MB | Commit: 0.15GB | Faults: 25605

### Stack Trace (bun.report)

Bun v1.1.13 ([`bd6a605`](<https://github.com/oven-sh/bun/tree/bd6a6051203ff577240615ff4e242270035a36b7>)) on windows x86_64 [InstallCommand]

**panic**: Internal assertion failure

- [`debug.zig:373`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/deps/zig/lib/std/debug.zig#L373>): `panicExtra`
- [`debug.zig:348`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/deps/zig/lib/std/debug.zig#L348>): `panic`
- [`output.zig:356`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/output.zig#L356>): `panic`
- [`bun.zig:3254`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/bun.zig#L3254>): `assertionFailure`
- [`lockfile.zig:1405`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/install/lockfile.zig#L1405>): `shouldPrintPackageInstall`
- [`lockfile.zig:1326`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/install/lockfile.zig#L1326>): `printInstalledWorkspaceSection`
- [`lockfile.zig:1611`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/install/lockfile.zig#L1611>): `print`
- [`filter_run.zig:431`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/cli/filter_run.zig#L431>): `install`
- [`cli.zig:1465`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/cli.zig#L1465>): `start`
- [`cli.zig:62`](<https://github.com/oven-sh/bun/blob/bd6a6051203ff577240615ff4e242270035a36b7/src/cli.zig#L62>): `start`

**Comment by wundersolutions-juanjo** [comment]
Yup, still reproducible;

Bun v1.1.17 ([`bb66bba`](<https://github.com/oven-sh/bun/tree/bb66bba1bf6d57983e7efef7ab706bce12137028>)) on windows x86_64 [InstallCommand]

**panic**: Internal assertion failure

- [`debug.zig:443`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/deps/zig/lib/std/debug.zig#L443>): `panicExtra`
- [`bun.zig:3252`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/bun.zig#L3252>): `assertionFailure`
- [`lockfile.zig:1438`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/install/lockfile.zig#L1438>): `shouldPrintPackageInstall`
- [`lockfile.zig:1359`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/install/lockfile.zig#L1359>): `printInstalledWorkspaceSection`
- [`lockfile.zig:1644`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/install/lockfile.zig#L1644>): `print`
- [`install.zig:13784`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/install/install.zig#L13784>): `installWithManager`
- [`filter_run.zig:431`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/cli/filter_run.zig#L431>): `install`
- [`cli.zig:1492`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/cli.zig#L1492>): `start`
- [`cli.zig:62`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/cli.zig#L62>): `start`
- [`main.zig:51`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/main.zig#L51>): `main`

**Comment by wundersolutions-juanjo** [comment]
I should also mention that after that crash, running the command again works fine, and that i'm pretty sure it's also reproducible on mac M1

**Comment by dylan-conway** [comment]
Fixed in bun v1.1.31 with #14461
